recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,756 @@
+        else:
+            rmdir(self, os.path.join(self._source_subfolder,"CMake","patches"))
+
+        # And apply our patches.  I do it here rather than in build, so I can repeatedly call build without applying patches.
+        apply_conandata_patches(self)
+
+
+
+    def source(self):
+        if self.options.use_source_from_git:
+            self.run("git clone -b release --single-branch " + self.git_url + " " + self._source_subfolder)
+            # note: we give the branch a name so we don't have detached heads
+            # TODO change to standard git and python chdir
+            git_hash = "v" + self.version
+            self.run("cd " + self._source_subfolder + " && git checkout -b branch-" + git_hash + " " + git_hash)
+        else:
+            get(**self.conan_data["sources"][self.version],
+                    strip_root=True,
+                    destination=self._source_subfolder,
+                    )
+
+        if self.no_copy_source:
+            self._patch_source()
+
+
+    def _third_party(self):
+        parties = {
+                # LEFT field:  target name for linking, will be used as TARGET::TARGET in package_info()
+                # RIGHT field: package/version to require
+                  "double-conversion": "double-conversion/3.2.0"
+                , "eigen":             "eigen/3.4.0"
+                , "exprtk":            "exprtk/0.0.1"
+                , "fmt":               "fmt/8.1.1"       # 9.1.0 release docs mention a PR - confirmed merged 8.1.0
+                , "freetype":          "freetype/2.11.1" # ... what? ... WORKED with 2022-05-06-freetype
+                , "glew":              "glew/2.2.0"
+                , "jsoncpp":           "jsoncpp/1.9.5"
+                # , "libharu": "libharu/2.3.0" -- use VTK's bundled version - heavily patched
+                , "lz4":               "lz4/1.9.3"
+                , "proj":              "proj/9.0.0" # if MAJOR version changes, update ThirdParty/libproj/CMakeLists.txt
+                , "pugixml":           "pugixml/1.12.1"
+                , "sqlite3":           "sqlite3/3.38.1"
+                , "utfcpp":            "utfcpp/3.2.1"
+                , "xz_utils":          "xz_utils/5.2.5" # VTK calls this lzma
+                }
+
+        # TODO figure out how we want to support modules...
+        # VTK already has an extensive module system, we would want to mirror or use it
+        # if self.options.module_hdf5:
+        # parties["hdf5"] = "hdf5/1.12.1"
+
+        if self.options.qt:
+            parties["qt"] = "qt/6.2.4"
+
+            # NOTE: could also try QT's offical QT
+            # self.requires("qtbase/6.2.4@qt/everywhere")
+
+        return parties
+
+
+    def requirements(self):
+        for pack in self._third_party().values():
+            self.requires(pack)
+
+
+    def build_requirements(self):
+        if self.options.rendering:
+            Apt(self).install([
```

Comment:
I don't think it will lbe allowed here.
in general, you might consider adding `xorg/system` and/or `opengl/system` to requirements

Reply:
Ok, I've added those to requirements()